packages/components/src/components/Forms/CheckboxGroup/CheckboxGroup.tsx

```diff
@@ -0,0 +1,102 @@
+import { View, ViewStyle } from 'react-native'
+import { spacing } from '@department-of-veterans-affairs/mobile-tokens'
+import React, { FC, Fragment } from 'react'
+
+import { Checkbox } from '../Checkbox/Checkbox'
+import { ComponentWrapper } from '../../../wrapper'
+import { Error, Header, Hint } from '../FormText'
+import { FormElementProps } from '../../../types/forms'
+import { Spacer } from '../../Spacer/Spacer'
+import { TextWithA11yAndValue } from '../../../types/common'
+import { useTheme } from '../../../utils'
+
+export type CheckboxGroupProps = FormElementProps & {
+  /** Array of checkbox options. Can be an array containing strings or objects if values or a11y overrides are needed */
+  items: string[] | TextWithA11yAndValue[]
+  /** Callback function that receives an updated array of selected values when checkboxes are pressed */
+  onSelectionChange: (selected: (string | number)[]) => void
+  /** Array of the labels or values (if provided) of currently selected checkboxes */
+  selectedItems: (string | number)[]
+}
+
+export const CheckboxGroup: FC<CheckboxGroupProps> = ({
+  items,
+  selectedItems,
+  error,
+  header,
+  hint,
+  onSelectionChange,
+  required,
+  tile,
+}) => {
+  const theme = useTheme()
+
+  const handleCheckboxChange = (value: string | number) => {
+    if (selectedItems.includes(value)) {
+      onSelectionChange(
+        selectedItems.filter((itemValue) => itemValue !== value),
+      )
+    } else {
+      onSelectionChange([...selectedItems, value])
+    }
+  }
+
+  /**
+   * Container styling
+   */
+  let containerStyle: ViewStyle = {
+    width: '100%',
+  }
+
+  if (error) {
+    containerStyle = {
+      ...containerStyle,
+      borderLeftWidth: spacing.vadsSpace2xs,
+      borderColor: theme.vadsColorFormsBorderError,
+      paddingLeft: spacing.vadsSpaceLg,
+    }
+  }
+
+  return (
+    <ComponentWrapper>
+      <View style={containerStyle}>
+        {header && (
+          <>
+            <Header text={header} required={required} />
+            <Spacer size="xs" />
+          </>
+        )}
+
+        {hint && (
+          <>
+            <Hint text={hint} />
+            <Spacer size="xs" />
+          </>
+        )}
+
+        {error && (
+          <>
+            <Error text={error} />
+            <Spacer size="xs" />
+          </>
+        )}
+        {items.map((item, index) => {
+          const value =
+            typeof item === 'object' ? item.value || item.text : item
```

Comment:
I'm finding this very confusing even after looking for a good amount of time. I assume it may have something to do with the form wrapper, but this "value" prop seems incredibly ambiguous. Is it supposed to be a key/id that can be used to identify which Checkbox item is being acted on instead of just the label?

Maybe related to the general not understanding, but it further seems unclear why this is needed vs the selectedItems depending solely on the labels which should always be present.

Reply:
The FormWrapper does use values for Radio Buttons, but I'm thinking beyond that to common use cases in forms. The app doesn't currently use any CheckboxGroups, but the options that show up in them or in Radio Buttons are likely to be a mapping of some list of values or IDs from an API, with separate display text.  Ultimately, these values are what are going to be passed back to the API in a form submit, not the labels. 

While it'd be simpler on our end to use labels passed to keep track of `selectedItems`, this would require an extra step on the user's end to then get the corresponding value to the selected label we pass them before submitting. So while it's a little more complex for us to support both, I think it'd be more convenient for devs.

Reply:
I think at a minimum we need something explaining this property better. "Value that may differ from provided text" is super vague and, with your clarification, I'm still not sure I follow the use case for this beyond CheckboxGroup and Radio.

Even that may be a stretch because I feel the uses are different:
- Radio:
    - Select one for one component
    - You pick one of many options (values) to store as data for one key/id/index/whatever in the app
- CheckboxGroup:
    - Select many for one or many component
        - You pick multiple of many options (values) to store as data for one key/id/index/whatever in the app OR 
        - You pick multiple of many options (values) to store as data for multiple separate key/id/index/whatever's in the app (each checkbox corresponds to app data)
    - On top of that, these are simultaneously keys themselves because they in turn store data: true/false of whether it is checked
    - So here these are simultaneously keys and values: keys for the item in the list (as being used from the component perspective), values for some app-level key or keys that cares about which boxes are checked

So I guess in fully thinking through it, they are more values than not coming from the consumer, but it still seems muddy in my mind that it's logical for CheckboxGroup/Radio to share, let alone potentially other components as well.

Maybe the answer is "trust me" from your research around FormWrapper, but as of this ticket I have skepticism we aren't trying to oversimplify saying this makes sense to be global vs just forms vs local to CheckboxGroup.

Reply:
Updated the comment for `value` to be a little clearer, but just to confirm, your main concern is less about the ability to provide `value`'s in the checkbox `items`, and more-so about whether this `TextWithA11yAndValue` type will be used across other form components?

I'm open to having them be local types to each CheckboxGroup and RadioButton, but this is kind of related to our coding style discussion. What should our criteria be for a type to be exported or in the shared type file like `forms.ts`? If it's used in at least X number of components? What is that number?

Reply:
> just to confirm, your main concern is less about the ability to provide `value`'s in the checkbox `items`, and more-so about whether this `TextWithA11yAndValue` type will be used across other form components?

Yeah. It's confusing to me as a global type since that implicitly means it should be useful to most components in my mind like the other types here which are relevant to basically every component that contains customizable text. As of this ticket, it's relevant to just CheckboxGroup and there's ambiguity in my mind if it does make sense to share 1:1 with Radio, let alone anything else after that.

> I'm open to having them be local types to each CheckboxGroup and RadioButton, but this is kind of related to our coding style discussion. What should our criteria be for a type to be exported or in the shared type file like `forms.ts`? If it's used in at least X number of components? What is that number?

Seems to me like `forms.ts` is a special case because the display logic itself is so repeatable, not sure we want to make a habit of abstracting typing too much. But to this ticket specifically: this is a global type currently that I'm not exactly clear should be shared at all so in my mind it should be local to the component and we can make it shared with the addition of Radio if it does make sense.